### PR TITLE
Gradle buildSrc cleanup

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Move `buildSrc` from namespace `toolkits.*` to `software.aws.toolkits.*`
- Move `validateLocalizedMessages` into buildSrc and convert to Kotlin + give it better messages

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                        it
+                    } else {
+                        LOG.warn(""""Invalid message in $filePath does not contain a '=': "$it"""")
+                        null
+                    }
+                }
+                .map { it.split("=").first() }
+                .reduce { item1, item2 ->
+                    if (item1 > item2) {
+                        throw GradleException("""localization file $filePath is not sorted:"$item1" > "$item2"""")
```
Comment (abrooksv) [inline buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt:37]: How difficult to collect all these in one go?

Comment (hunterwerlla) [reply buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt]: Sure, updated

File: build.gradle [hunk 2]
```diff
@@ -278,31 +279,8 @@ task ktlint(type: JavaExec, group: "verification") {
...
-            .filter({ item -> !item.isEmpty() })
-            .reduce({ item1, item2 ->
-                if (item1 > item2) {
-                    throw new GradleException("localization file is not sorted:" + item1 + " > " + item2)
-                }
-
-                return item2
-            })
-    }
+tasks.register('validateLocalizedMessages', ValidateMessages) {
```
Comment (rli) [inline build.gradle:282]: do we want to keep the "verification" group identifier?

Comment (hunterwerlla) [reply build.gradle:282]: Good catch I thought I readded that, fixed

File: buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskAction
+import org.gradle.language.base.plugins.LifecycleBasePlugin.VERIFICATION_GROUP
+import java.io.File
+
+open class ValidateMessages : DefaultTask() {
+    @InputFiles
+    lateinit var paths: List<File>
```
Comment (abrooksv) [inline buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt:17]: You can avoid lateineit using `project.objects.directoryProperty()`

File: buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    item2
+                }
+            if (hasError) {
+                throw GradleException("$filePath has one or more out of order items!")
+            }
+        }
+    }
+
+    private companion object {
+        val LOG: Logger = Logging.getLogger(ValidateMessages::class.java)
```
Comment (abrooksv) [inline buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt:56]: Oh, dont make loggers this way.

Use the `Task#getLogger()` at the log site

Comment (hunterwerlla) [reply buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt]: Changed

File: buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt (hunk 3 above)
Comment (hunterwerlla) [reply buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt]: Changed (to listProperty)

**Pull request merged**